src/box/memtx_engine.cc

```diff
@@ -2032,16 +2034,19 @@ memtx_tuple_new_raw_impl(struct tuple_format *format, const char *data,
 		goto end;
 	}
 
-	while ((tuple = MemtxAllocator<ALLOC>::alloc_tuple(total)) == NULL) {
+	while ((block = MemtxAllocator<ALLOC>::alloc(data_size,
+						     make_compact)) == NULL) {
 		bool stop;
 		memtx_engine_run_gc(memtx, &stop);
 		if (stop)
 			break;
 	}
-	if (tuple == NULL) {
+	if (block == NULL) {
 		diag_set(OutOfMemory, total, "slab allocator", "memtx_tuple");
 		goto end;
 	}
+
+	tuple = memtx_block_get_tuple(block);
 	tuple_create(tuple, 0, tuple_format_id(format),
```

Comment:
It'd be nice to avoid overwriting the tuple size (AFAIU it was initialized by the memtx allocator). Maybe, split the `tuple_create` function and introduce some kind of "base" variant, which wouldn't overwrite the size?